### PR TITLE
testing please do not merge

### PR DESCRIPTION
**Pull request by tuan-phan**

### CODE REVIEW

File: lib/gooddata/test.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# encoding: UTF-8
+
+module GoodData
+  module Connectors
+    module Metadata
+      class MetadataContext
+        attr_accessor :default_folder, :account_id, :token, :data_source, :entity, :customer, :integrator
```
Comment (yenkins) [inline lib/gooddata/test.rb]: Is a writable attribute (Attribute)

File: lib/gooddata/test.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            object,
+            TimeHelper.year(time),
+            TimeHelper.month(time),
+            TimeHelper.day(time),
+            path
+          )
+        end
+
+        def construct_path(level, path)
+          case level
```
Comment (yenkins) [inline lib/gooddata/test.rb]: Is controlled by argument 'level' (ControlParameter)

File: lib/gooddata/test.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2018, GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# encoding: UTF-8
+
+module GoodData
+  module Connectors
+    module Metadata
+      class MetadataContext
```
Comment (yenkins) [inline lib/gooddata/test.rb]: Has no descriptive comment (IrresponsibleModule)

File: lib/gooddata/test.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            'cache',
+            path
+          )
+        end
+
+        def construct_full_metadata_path(entity, path, time = Time.new.utc)
+          construct_full_path('metadata', entity, path, time)
+        end
+
+        def construct_full_path(type, object, path, time = Time.new.utc)
```
Comment (yenkins) [inline lib/gooddata/test.rb]: Has 4 parameters (LongParameterList)

File: lib/gooddata/test.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            TimeHelper.year(time),
+            TimeHelper.month(time),
+            TimeHelper.day(time),
+            path
+          )
+        end
+
+        def construct_path(level, path)
+          case level
+            when 1
```
Comment (yenkins) [inline lib/gooddata/test.rb]: Style/CaseIndentation: Indent `when` as deep as `case`.

File: lib/gooddata/test.rb [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+              File.join(default_folder, account_id, path)
+            when 2
```
Comment (yenkins) [inline lib/gooddata/test.rb]: Style/CaseIndentation: Indent `when` as deep as `case`.

File: lib/gooddata/test.rb [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+              File.join(default_folder, account_id, token, path)
+            when 3
```
Comment (yenkins) [inline lib/gooddata/test.rb]: Style/CaseIndentation: Indent `when` as deep as `case`.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/94/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/913/ : FAILURE in 46s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1181/ : SUCCESS in 1m 54s

**Comment by tuan-phan** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/95/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/914/ : FAILURE in 40s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1182/ : SUCCESS in 39s

**Comment by tuan-phan** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/96/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1360/ : SUCCESS in 55m 53s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1123/ : SUCCESS in 33m 36s

**Comment by tuan-phan** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/113/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/939/ : FAILURE in 2m 04s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1208/ : SUCCESS in 2m 02s

**Pull request closed without merging**